### ISSUE TITLE
.pcs read and write error

### ISSUE THREAD

**Issue by VinceDT** [automl/SMAC3#489]
#### Description
.pcs file read and write does not work

#### Steps/Code to Reproduce
```
#!/usr/bin/env python

from smac.configspace import (ConfigurationSpace,
                              Configuration)
from ConfigSpace.hyperparameters import (CategoricalHyperparameter,
                                         UniformFloatHyperparameter,
                                         UniformIntegerHyperparameter, )
from ConfigSpace.read_and_write import pcs
from smac.utils.io.output_writer import OutputWriter
from smac.utils.io.input_reader import InputReader



cs = ConfigurationSpace()
hyp = UniformFloatHyperparameter('A',
                                 0.0,
                                 1.0,
                                 default_value=0.5)
cs.add_hyperparameters([hyp])

o = OutputWriter()
o.save_configspace(cs, 'configspace.pcs', 'pcs_new')

i = InputReader()
cs = i.read_pcs_file('configspace.pcs')
# first error, empty output, parameter is not read in, in read_pcs_file the file is not opened
# with open(fn) as file:
#     file_string = file.readlines()
print(cs)

# Optional solution also not working: Seems like the pcs format changed at some point
# If 'real' is removed it works
with open('configspace.pcs') as file:
    file_string = file.readlines()
    cs = pcs.read(file_string)
print(cs)
```

#### Versions
0.10.0

Thanks!

**Comment by shukon** [comment]
So in general, there are three formats for the configuration-space. `pcs`, `pcs_new` and `json`. We recommend to use json wherever possible. However, if you use `pcs`/`pcs_new` you will have to load with the same function that you save with.

(1) There seems to be a bug here. I don't think the function is actually used anywhere, so we have to either update or delete it. It currently only reads in config-spaces in the **old pcs**-format
My proposal: Update the function in the input-reader to be more like the [code we actually use in the cmd-reader](https://github.com/automl/SMAC3/blob/b5c398655a5ff4e15ecfd3e23afda5303cdcbc06/smac/utils/io/cmd_reader.py#L139-L149).

(2) Since you are saving the configfile in the `pcs_new` in the line `o.save_configspace(cs, 'configspace.pcs', 'pcs_new')`, you will also have to load it using the `pcs_new`-function. Solution: change your import from `from ConfigSpace.read_and_write import pcs` to `from ConfigSpace.read_and_write import pcs_new as pcs` and you're ready to go!

However I think we could increase the informativity of the error-message. Currently it just throws

```
Traceback (most recent call last):
  File "reproduce.py", line 35, in <module>
    cs = pcs.read(file_string)
  File "/home/shuki/virtual_environments/smac/lib/python3.6/site-packages/ConfigSpace/read_and_write/pcs.py", line 237, in read
    raise NotImplementedError("Could not parse: %s" % line)
NotImplementedError: Could not parse: A real [0.0, 1.0] [0.5]
```

**Comment by VinceDT** [comment]
Great, thanks a lot for the clarification.

**Comment by mfeurer** [comment]
Closed via #489.
